iothub/device/iot-device-client/src/test/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttTwinTest.java

```diff
@@ -904,6 +904,33 @@ public void receiveDoesNotSetVersionForDesiredPropNotifRespIfNotFound() throws T
         }
     }
 
+    @Test
+    public void receiveSetQosForDesiredProp() throws TransportException
+    {
+        final byte[] actualPayload = "NotificationResponseDataContainingDesiredPropertiesDocument".getBytes(StandardCharsets.UTF_8);
+        final String expectedTopic = "$iothub/twin/PATCH/properties/desired/";
+        IotHubTransportMessage receivedMessage = null;
+        int expectedQualityOfService = 2;
```

Comment:
Small nitpick to make this unit test closer to what an actual received twin message would look like.

```suggestion
        int expectedQualityOfService = 0;
```

Reply:
In other case I would agree - I'm all for making test data as close to actual production data as possible. But doing so, I would made this test passing regarding of production code change - QOS is held as `int` which during object construction is set to default value: 0.

First I created failing test, and the fix production making test green - this way I knew that test is actually checking case I wanted to cover.